src/combinator.rs

```diff
@@ -625,3 +642,41 @@ impl<I: Clone, O, A: Parser<I, O, Error = E>, U: Clone, E: Error<I>> Parser<I, U
     #[inline]
     fn parse_inner_silent(&self, d: &mut Silent, s: &mut StreamOf<I, E>) -> PResult<I, U, E> { #[allow(deprecated)] self.parse_inner(d, s) }
 }
+
+#[cfg(test)]
+mod tests {
+    use error::Simple;
+
+    use super::*;
+
+    #[test]
+    fn separated_by_at_least() {
+        let parser = just::<_, Simple<char>>('-')
+            .separated_by(just('.'))
+            .at_least(3);
+        
+        assert_eq!(parser.parse("-.-.-"), Ok(vec!['-', '-', '-']));
```

Comment:
Perhaps an error-generating pattern here too like `-.-.`? For me, it's quite important that we make sure that this continues to correctly interpret trailing/leading patterns as errors when used as `at_least` and without `allow_trailing`/`allow_leading`.

Reply:
Good point, I'll add that.